egui/src/widgets/plot/items/values.rs

```diff
@@ -140,22 +147,45 @@ impl Default for Orientation {
 
 // ----------------------------------------------------------------------------
 
-#[derive(Default)]
-pub struct Values {
-    pub(super) values: Vec<Value>,
-    generator: Option<ExplicitGenerator>,
+pub enum PlotPoints<'v> {
+    Borrowed(&'v [PlotPoint]),
+    Owned(Vec<PlotPoint>),
+    Generator(ExplicitGenerator),
 }
 
-impl Values {
-    pub fn from_values(values: Vec<Value>) -> Self {
-        Self {
-            values,
-            generator: None,
-        }
+impl From<[f64; 2]> for PlotPoints<'_> {
+    fn from(coordinate: [f64; 2]) -> Self {
+        Self::new(vec![coordinate])
     }
+}
+
+impl From<Vec<[f64; 2]>> for PlotPoints<'_> {
+    fn from(coordinates: Vec<[f64; 2]>) -> Self {
+        Self::new(coordinates)
+    }
+}
 
-    pub fn from_values_iter(iter: impl Iterator<Item = Value>) -> Self {
-        Self::from_values(iter.collect())
+impl<'v> From<&'v [[f64; 2]]> for PlotPoints<'v> {
+    fn from(coordinates: &'v [[f64; 2]]) -> Self {
+        Self::from_slice(coordinates)
+    }
+}
+
+impl<'v> PlotPoints<'v> {
+    pub fn new(points: Vec<[f64; 2]>) -> Self {
+        Self::Owned(bytemuck::cast_vec(points))
+    }
+
+    pub fn from_slice(points: &'v [[f64; 2]]) -> Self {
+        Self::Borrowed(bytemuck::cast_slice(points))
+    }
```

Comment:
`bytemuck` is nice, but it also makes compiling egui from scratch take ~twice as long (because of the `syn` crate). I'd rather keep the number of required dependencies low, so let's make `bytemuck` opt-in.

I think it's good to prefix these with `#[cfg(feature = "bytemuck")]`

Reply:
Ah good point. I removed the dependency again. I also removed `PlotPoints::Borrowed` for now because I couldn't find a way to to make it work with arbitrary lifetimes. Once somebody finds a way to get that working, it will make sense to add bytemuck again.